### ISSUE TITLE
Log multi-task loss weights (log_vars) to wandb for diagnostics

### ISSUE THREAD

**Issue by Polichinel** [views-platform/views-hydranet#48]
## Context

The `MultiTaskLoss` (Kendall et al. 2018) learns per-task σ values that balance regression and classification losses. These are `nn.Parameter` values (`log_vars`) that the optimizer adjusts during training — but we never see what they converge to.

The learnable sigma sweep (issue #44) revealed that training loss curves show classification (focal) much more volatile than regression (Tobit), and the total multi-task loss appears driven primarily by regression. Whether this is a convergence issue (the task balancer hasn't found equilibrium) or a scale issue (correct behavior given the loss magnitudes) is impossible to diagnose without seeing the learned weights.

## Proposed Change

Log `log_vars` and derived weights to wandb once per lesson, same pattern as sigma logging (ADR-055):

```python
wandb.log({
    f"mtl_weight/{name}": coeff.item()
    for name, coeff in zip(target_names, coeffs)
})
wandb.log({
    f"mtl_log_var/{name}": lv.item()
    for name, lv in zip(target_names, multitaskloss_instance.log_vars)
})
```

This requires passing target names into the logging context — currently `MultiTaskLoss` only knows task indices, not names.

## Expected Diagnostic Value

- If log_vars converge to very different values for reg vs cls → the balancer is working, scale mismatch is handled
- If log_vars are still drifting at lesson 80 → need longer training or different learning rate for task weights
- If cls log_vars are much larger than reg → confirms the balancer is down-weighting volatile classification loss (correct behavior)

**Comment by Polichinel** [comment]
Implemented in PR #51 (commit 0258c0b). Per-target `mtl_log_var/{target_name}` logged to wandb once per lesson after optimizer step. First results from production sweep show all log_vars stuck at 0.000 — the Kendall balancer is not learning task weights (likely AdamW weight_decay=0.1 regularizing log_vars back to zero). This is a finding in itself, documented in the sweep results.
